py/strato/monitors/exceptionfinder.py

```diff
@@ -0,0 +1,213 @@
+import json
+from strato.common.log import config as logconfig
+import os
+import datetime
+import re
+import calendar
+import yaml
+import logging
+import time
+from collections import defaultdict
+
+FILEBEAT_CONFIG_FILE_PATH = 'tools/monitoring/filebeat.yml'
+OUTPUT_PATH = os.path.join(logconfig.LOGS_DIRECTORY, 'exceptions.stratolog')
+LOGS_WITH_COMMON_FORMAT = ['consul', 'nova', 'neutron', 'keystone']
+LOGS_DISTRIBUTION = defaultdict(int)
+LOGS_TYPES_FOR_ADJUSTMENT = list(LOGS_WITH_COMMON_FORMAT) + ['vm-console', 'mancala-bin']
+
+
+def start(hosts):
+    try:
+        filebeatinstall.installFilebeat(hosts, FILEBEAT_CONFIG_FILE_PATH)
+    except:
+        logging.warning('Failed to install filebeat')
+
+
+def stop(hosts):
+    try:
+        filebeatinstall.stopFilebeat(hosts)
+        createExceptionFile(hosts)
+        printLogsDistribution()
+    except:
+        logging.warning('Failed to create exception file')
+
+
+def createExceptionFile(hosts):
+    exceptions = []
+    filebeatConfig = yaml.load(open(FILEBEAT_CONFIG_FILE_PATH, 'r').read())
+    exceptionFileDir = filebeatConfig['output']['file']['path']
+    exceptionFileBaseName = filebeatConfig['output']['file']['filename']
+    for host in hosts:
+        timeGap = _findTimezoneOffset(host)
+        for fileName in _listAllExceptionFilesOnHost(host, exceptionFileDir, exceptionFileBaseName):
+            try:
+                filePath = os.path.join(exceptionFileDir, fileName)
+                hostExceptions = host.ssh.ftp.getContents(filePath).strip().split('\n')
+                exceptions.extend([_formaliseExceptionLog(line, host.name, timeGap) for line in hostExceptions])
+                exceptions.extend(_getBeforeAndAfterRunLogs(host, timeGap))
+            except:
+                logging.warning('failed to collect exception log from %s' % host.name)
+                continue
+    try:
+        sortedList = sorted(exceptions, key=lambda k: k['created'])
+    except:
+        logging.warning("Failed to parse some of the lines. Omitting the failed lines")
+        sortedList = sorted([line for line in exceptions if line], key=lambda k: k['created'])
```

Comment:
yeah, that's why i've added that except.. the "if line" removes all the None-s.
however, if it gets here, it's a bug on my part and i want to know when it happens, so that i'll know that i need to debug it